shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -405,30 +426,186 @@ private Drawable getSplashScreenDrawableFromActivityTheme() {
     }
   }
 
+  @Override
+  public void onStart() {
+    super.onStart();
+    Log.d(TAG, "onStart()");
+    flutterEngine.getSystemChannels().lifecycle.appIsInactive();
+  }
+
+  public void onPostResume() {
+    Log.d(TAG, "onPostResume()");
+    // TODO(mattcarroll): what does 'updateAccessibilityFeatures' do? why is it called here?
+    flutterView.updateAccessibilityFeatures();
+    // TODO(mattcarroll): why does the platform plugin care about "post resume"?
+    platformPlugin.onPostResume();
+    // TODO(mattcarroll): why does the lifecycle wait for "post resume" to say we're resumed?
+    flutterEngine.getSystemChannels().lifecycle.appIsResumed();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    Log.d(TAG, "onPause()");
+    flutterEngine.getSystemChannels().lifecycle.appIsInactive();
+  }
+
+  @Override
+  public void onStop() {
+    super.onStop();
+    Log.d(TAG, "onStop()");
+    flutterEngine.getSystemChannels().lifecycle.appIsPaused();
+  }
+
+  @Override
+  public void onDestroyView() {
+    super.onDestroyView();
+    Log.d(TAG, "onDestroyView()");
+    flutterView.detachFromFlutterEngine();
+
+    // TODO(mattcarroll): what does 'detach' refer to?  The Activity? The UI? JNI?....
```

Comment:
Drops the registry's ref to the FlutterView and Activity, plus does some cleanup of the message channel.

Reply:
I think we need to revisit the whole concept of attach() and detach() for a PluginRegistry.  We can cover that in the plugin meeting next week.

There shouldn't really be any reason why a FlutterView is special.  FlutterView should just be an Android View that happens to implement what is necessary to render Flutter.  So the notion that a PluginRegistry can be attached to a FlutterView, let alone that it *must* be attached to a FlutterView, runs counter to what I'd like to see in the API.

Reply:
On the iOS side, I'm making the methods related to this on FlutterViewController just pass through to FlutterEngine.  Could that work here?

Reply:
Which methods are you referring to?

Reply:
Ahhh looks like you're doing that too.

I'm not sure but I'd imagine detach here is notifying the engine it can tear down the rasterizer?

Reply:
Ahhh looks like you're doing that too.

I'm not sure but I'd imagine detach here is notifying the engine it can tear down the rasterizer?

Reply:
No, it's totally unrelated to rendering.  `detach()` is being called on the PluginRegistry, not the FlutterEngine.  This is why I think we need to revisit the whole relationship of PluginRegistry vs surrounding objects.

Reply:
Are you talking about just plugins here or platform channels in general?

My 2 cents:  Plugins (and registry etc) are indeed too heavy weight to be attached to views. However, I would not like to make platform channels only scoped to the engine. 

We want to give developers options on how they want to implement their Flutter features:
- "deep" Flutter features (model, UI in Dart and only talk to native _services_)
- "shallow" ones where Flutter is just UI and everything else is native.

In the latter, having platform channels specific to a view is important IMO.